designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/ScenarioStatistics.scala

```diff
@@ -0,0 +1,178 @@
+package pl.touk.nussknacker.ui.statistics
+
+import cats.implicits.toFoldableOps
+import pl.touk.nussknacker.engine.api.component.{ComponentType, ProcessingMode}
+import pl.touk.nussknacker.engine.api.deployment.simple.SimpleStateStatus
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+import pl.touk.nussknacker.restmodel.component
+import pl.touk.nussknacker.ui.process.processingtype.DeploymentManagerType
+import pl.touk.nussknacker.ui.process.repository.DbProcessActivityRepository
+
+import java.time.Instant
+
+object ScenarioStatistics {
+
+  def getGeneralStatistics(scenariosInputData: List[ScenarioStatisticsInputData]): Map[String, String] = {
+    if (scenariosInputData.isEmpty) {
+      Map.empty
+    } else {
+      val scenariosCount = scenariosInputData.count(!_.isFragment)
+      //        Nodes stats
+      val sortedNodes  = scenariosInputData.map(_.nodesCount).sorted
+      val nodesMedian  = calculateMedian(sortedNodes)
+      val nodesAverage = sortedNodes.sum / scenariosInputData.length
+      val nodesMax     = sortedNodes.head
+      val nodesMin     = sortedNodes.last
+      //        Category stats
+      val categoriesCount = scenariosInputData.map(_.scenarioCategory).toSet.size
+      //        Version stats
+      val sortedVersions  = scenariosInputData.map(_.scenarioVersion.value).sorted
+      val versionsMedian  = calculateMedian(sortedVersions)
+      val versionsAverage = sortedVersions.sum / scenariosInputData.length
+      val versionsMax     = sortedVersions.head
+      val versionsMin     = sortedVersions.last
+      //        Author stats
+      val authorsCount = scenariosInputData.map(_.createdBy).toSet.size
+      //        Fragment stats
+      val fragments        = scenariosInputData.map(_.fragmentsUsedCount).sorted
+      val fragmentsMedian  = calculateMedian(fragments)
+      val fragmentsAverage = fragments.sum / scenariosCount
```

Comment:
Fragments count? Do we have test for it?

Reply:
I changed it to fragmentsUsedCount so it won't be mistaken with amount of created fragments